app/lib/email_alert_list_title_builder.rb

```diff
@@ -0,0 +1,55 @@
+class EmailAlertListTitleBuilder
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(filter:, subscription_list_title_prefix:, facets:)
+    @filter = filter
+    @subscription_list_title_prefix = subscription_list_title_prefix
+    @facets = facets
+  end
+
+  def call
+    join_fragment = "in the following #{'category'.pluralize(selected_facet_choice_names.count)}"
+    "#{subscription_list_title_prefix.strip} #{join_fragment}: #{comma_separated_facet_choices}"
+  end
+
+private
+
+  attr_accessor :filter, :subscription_list_title_prefix, :facets
+
+  def comma_separated_facet_choices
+    selected_facet_choice_names.map { |choice_name| "'#{choice_name}'" }.join(", ")
+  end
+
+  def selected_facet_choice_names
+    @selected_facet_choice_names ||= begin
+      overwritten_filters = filter.map do |selected_facet_id, selected_facet_choices|
+        selected_facet_choices.map do |selected_facet_choice_id|
+          overwritten_name = facet_name_overrides.fetch(selected_facet_id, {})[selected_facet_choice_id]
+          overwritten_name || fetch_facet_choice_name(selected_facet_id, selected_facet_choice_id)
```

Comment:
Could these two lines be simplified into one without any assignment?

```suggestion
          facet_name_overrides.fetch(selected_facet_id, {})[selected_facet_choice_id] || fetch_facet_choice_name(selected_facet_id, selected_facet_choice_id)
```

Reply:
That line looked long and hard to read. Instead, I've got it calling a `fetch_facet_choice_name` method, which in turn calls `fetch_facet_choice_name_from_overrides` or `fetch_facet_choice_name_from_data`. Looks good to me now, please double-check 👍